The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8396109680/job/22996802794) [(plain)](https://github.com/rust-lang/rust/commit/86a391c3f0376e9d36bb0cd60945ac1ef1ea6e61/checks/22996802794/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
.................................................................

failures:

---- library/core/src/../../portable-simd/crates/core_simd/src/vector.rs - core_simd::vector::Simd<T,N>::load_or (line 347) stdout ----
error: unused import: `Mask`
error: doctest failed, to rerun pass `-p core --doc`
  |
  |
7 | use simd::{Simd, Mask};
  |
note: the lint level is defined here
 --> library/core/src/../../portable-simd/crates/core_simd/src/vector.rs:345:9
  |
---
---- library/core/src/../../portable-simd/crates/core_simd/src/vector.rs - core_simd::vector::Simd<T,N>::load_or_default (line 324) stdout ----
error: unused import: `Mask`
##[error] --> library/core/src/../../portable-simd/crates/core_simd/src/vector.rs:328:18
  |
7 | use simd::{Simd, Mask};
  |
note: the lint level is defined here
 --> library/core/src/../../portable-simd/crates/core_simd/src/vector.rs:322:9
  |
```

</details>

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -141,6 +141,11 @@ jobs:
       - name: Test (release)
         run: cargo test --verbose --target=${{ matrix.target }} --release
 
+      - name: Generate docs
+        run: cargo doc --verbose --target=${{ matrix.target }}
+        env:
+          RUSTDOCFLAGS: -Dwarnings
+
   wasm-tests:
     name: "wasm (firefox, ${{ matrix.name }})"
     runs-on: ubuntu-latest

--- a/crates/core_simd/src/lib.rs
+++ b/crates/core_simd/src/lib.rs
@@ -13,6 +13,7 @@
     simd_ffi,
     staged_api,
     strict_provenance,
+    prelude_import,
     ptr_metadata
 )]
 #![cfg_attr(
@@ -42,6 +43,10 @@
 #![allow(internal_features)]
 #![unstable(feature = "portable_simd", issue = "86656")]
 //! Portable SIMD module.
+
+#[prelude_import]
+#[allow(unused_imports)]
+use core::prelude::v1::*;
 
 #[path = "mod.rs"]
 mod core_simd;

--- a/crates/core_simd/src/lib.rs
+++ b/crates/core_simd/src/lib.rs
@@ -40,6 +40,7 @@
 )]
 #![warn(missing_docs, clippy::missing_inline_in_public_items)] // basically all items, really
 #![deny(unsafe_op_in_unsafe_fn, clippy::undocumented_unsafe_blocks)]
+#![doc(test(attr(deny(warnings))))]
 #![allow(internal_features)]
 #![unstable(feature = "portable_simd", issue = "86656")]
 //! Portable SIMD module.

--- a/crates/core_simd/src/vector.rs
+++ b/crates/core_simd/src/vector.rs
@@ -325,7 +325,7 @@ where
     /// # #![feature(portable_simd)]
     /// # #[cfg(feature = "as_crate")] use core_simd::simd;
     /// # #[cfg(not(feature = "as_crate"))] use core::simd;
-    /// # use simd::{Simd, Mask};
+    /// # use simd::Simd;
     /// let vec: Vec<i32> = vec![10, 11];
     ///
     /// let result = Simd::<i32, 4>::load_or_default(&vec);
@@ -348,7 +348,7 @@ where
     /// # #![feature(portable_simd)]
     /// # #[cfg(feature = "as_crate")] use core_simd::simd;
     /// # #[cfg(not(feature = "as_crate"))] use core::simd;
-    /// # use simd::{Simd, Mask};
+    /// # use simd::Simd;
     /// let vec: Vec<i32> = vec![10, 11];
     /// let or = Simd::from_array([-5, -4, -3, -2]);
     ///
@@ -695,7 +695,7 @@ where
     ///
     /// # Safety
     ///
-    /// Memory addresses for element are calculated [`core::ptr::wrapping_offset`] and
+    /// Memory addresses for element are calculated [`pointer::wrapping_offset`] and
     /// each enabled element must satisfy the same conditions as [`core::ptr::write`].
     #[inline]
     pub unsafe fn store_select_ptr(self, ptr: *mut T, enable: Mask<<T as SimdElement>::Mask, N>) {

--- a/crates/core_simd/src/vector.rs
+++ b/crates/core_simd/src/vector.rs
@@ -4,6 +4,7 @@ use crate::simd::{
     ptr::{SimdConstPtr, SimdMutPtr},
     LaneCount, Mask, MaskElement, SupportedLaneCount, Swizzle,
 };
+use core::convert::{TryFrom, TryInto};
 
 /// A SIMD vector with the shape of `[T; N]` but the operations of `T`.
 ///

